openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py

```diff
@@ -0,0 +1,590 @@
+                        create_uid, write_uid, create_date, write_date)
+                    SELECT {parent_id}, parent_path, name, code_prefix_start,
+                        code_prefix_end, {company_id}, create_uid,
+                        write_uid, create_date, write_date
+                    FROM account_group
+                    WHERE id = {id}
+                    RETURNING id
+                    """.format(
+                            id=group.id,
+                            company_id=company.id,
+                            parent_id=group.parent_id
+                            and relation_dict[company.id][group.parent_id.id].id
+                            or "NULL",
+                        )
+                    )
+                    new_group = env["account.group"].browse(env.cr.fetchone())
+                    relation_dict[company.id][group.id] = new_group
+            for account in accounts.filtered(
+                lambda a: a.group_id != relation_dict[a.company_id.id][a.group_id.id]
+            ):
+                account.group_id = relation_dict[account.company_id.id][
+                    account.group_id.id
+                ]
+    if compute_parent_path:
+        env["account.group"]._parent_store_compute()
+
+
+def fill_company_account_journal_suspense_account_id(env):
+    companies = env["res.company"].search([("chart_template_id", "!=", False)])
+    for company in companies:
+        chart = company.chart_template_id
+        account = chart._create_liquidity_journal_suspense_account(
+            company, chart.code_digits
+        )
+        company.account_journal_suspense_account_id = account
+    journals = env["account.journal"].search(
+        [("type", "in", ("bank", "cash")), ("company_id", "in", companies.ids)]
+    )
+    journals._compute_suspense_account_id()
+
+
+def fill_statement_lines_with_no_move(env):
+    stl_dates = {}
+    env.cr.execute(
+        """
+        SELECT id, %s
+        FROM account_bank_statement_line
+        WHERE move_id IS NULL"""
+        % (openupgrade.get_legacy_name("date"),)
+    )
+    for stl_id, stl_date in env.cr.fetchall():
+        stl_dates[stl_id] = stl_date
+    st_lines = env["account.bank.statement.line"].browse(list(stl_dates.keys()))
+    for st_line in st_lines.with_context(check_move_validity=False):
+        move = env["account.move"].create(
+            {
+                "name": "/",
+                "date": stl_dates[st_line.id],
+                "statement_line_id": st_line.id,
+                "move_type": "entry",
+                "journal_id": st_line.statement_id.journal_id.id,
+                "company_id": st_line.statement_id.company_id.id,
+                "currency_id": st_line.statement_id.journal_id.currency_id.id
+                or st_line.statement_id.company_id.currency_id.id,
+            }
+        )
+        st_line.move_id = move
+        st_line._synchronize_to_moves(
```

Comment:
I don't know how it is possible, but I have some old data which do not have the field date set:
This query returns 9 records from 2018 (when I was probably using an older version) `select id,name from account_move_line where date is NULL;`.
So the real source if the issue might be an old version of an OpenUpgrade.

Anyway, I will fix manually these few records: 
`update account_move_line set date=create_date where date is NULL;`

Reply:
Actually, it's even more strange, because in the source database in v13, I did not have account_move_line without dates. I ll keep digging...

Reply:
I confirm the error:
account: error in migration script /home/alexis/new_boite/dev/openupgrade4/openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py: You cannot add/modify entries prior to and inclusive of the lock date 31/12/2020
In the backtrace, the first line for openupgrade_script is:
```
File "/home/alexis/new_boite/dev/openupgrade4/openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py", line 544, in fill_account_payment_with_no_move
    payment._synchronize_to_moves(
```

Reply:
Ok, I understand, you are commenting the wrong line, so I got confused. `_synchronize_to_moves` is called two times, one for statement lines in `fill_statement_lines_with_no_move` and the other for payments in `fill_account_payment_with_no_move`. You are commenting in `fill_statement_lines_with_no_move` but you are having the error in `fill_account_payment_with_no_move`.

Reply:
I was able to reproduce the error of _check_fiscalyear_lock_date with a client database, so it should be a common issue. So I added query to avoid having moves without date.